specs/Workers.md

```diff
@@ -0,0 +1,1228 @@
+  /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings).
+  [propget] HRESULT Name([out, retval] LPWSTR* value);
+}
+```
+
+```c# (but really MIDL3)
+runtimeclass CoreWebView2DedicatedWorkerCreatedEventArgs;
+runtimeclass CoreWebView2DedicatedWorker;
+runtimeclass CoreWebView2ServiceWorkerRegistration;
+runtimeclass CoreWebView2ServiceWorkerRegisteredEventArgs;
+runtimeclass CoreWebView2ServiceWorkerManager;
+runtimeclass CoreWebView2ServiceWorker;
+runtimeclass CoreWebView2SharedWorkerManager;
+runtimeclass CoreWebView2SharedWorkerCreatedEventArgs;
+runtimeclass CoreWebView2SharedWorker;
+
+namespace Microsoft.Web.WebView2.Core
+{
+    [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2_25")]
+    {
+        event Windows.Foundation.TypedEventHandler<CoreWebView2, CoreWebView2DedicatedWorkerCreatedEventArgs> DedicatedWorkerCreated;
+    }
+
+    // ...
+    runtimeclass CoreWebView2DedicatedWorkerCreatedEventArgs
+    {
+        CoreWebView2DedicatedWorker Worker { get; };
+    }
+
+    runtimeclass CoreWebView2DedicatedWorker
+    {
+        String Name { get; };
+
+        String ScriptUri { get; };
+
+        event Windows.Foundation.TypedEventHandler<CoreWebView2DedicatedWorker, IInspectable> Destroyed;
+    }
+
+    enum CoreWebView2ServiceWorkerState
+    {
+        New = 0,
+        Installing = 1,
+        Installed = 2,
+        Activating = 3,
+        Activated = 4,
+        Redundant = 5,
+    };
+
+    [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2Profile9")]
+    {
+        CoreWebView2ServiceWorkerManager ServiceWorkerManager { get; };
+
+        CoreWebView2SharedWorkerManager SharedWorkerManager { get; };
+    }
+
+    runtimeclass CoreWebView2ServiceWorkerManager
+    {
+        event Windows.Foundation.TypedEventHandler<CoreWebView2ServiceWorkerManager, CoreWebView2ServiceWorkerRegisteredEventArgs> ServiceWorkerRegistered;
+
+        Windows.Foundation.IAsyncOperation<IVectorView<CoreWebView2ServiceWorkerRegistration>> GetServiceWorkerRegistrationsAsync();
+
+        Windows.Foundation.IAsyncOperation<CoreWebView2ServiceWorkerRegistration> GetServiceWorkerRegistrationAsync(String Scope);
+    }
+
+    runtimeclass CoreWebView2ServiceWorkerRegisteredEventArgs
+    {
+        CoreWebView2ServiceWorkerRegistration ServiceWorkerRegistration { get; };
+    }
+
+    runtimeclass CoreWebView2ServiceWorkerRegistration
+    {
+        String Scope { get; };
+
+        event Windows.Foundation.TypedEventHandler<CoreWebView2ServiceWorkerRegistration, IInspectable> Destroyed;
+
+        Windows.Foundation.IAsyncOperation<CoreWebView2ServiceWorker> GetServiceWorkerAsync();
+    }
+
+    runtimeclass CoreWebView2ServiceWorker
+    {
+        String ScriptUri { get; };
+
+        CoreWebView2ServiceWorkerState State { get; };
+
+        event Windows.Foundation.TypedEventHandler<CoreWebView2ServiceWorker, IInspectable> Destroyed;
```

Comment:
I've added this to notify apps that either the SW is unregistered, or we are destroying the object during profile destruction and such but I'm not sure if helps devs much. @david-risney What do you having of having this event on SWR? Do we notify apps that we are destroying the object and such?